broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -276,6 +279,35 @@ private void transitionToLeader(final CompletableActorFuture<Void> transitionCom
   }
 
   private ActorFuture<Void> basePartitionInstallation() {
+    final ActorFuture installFuture = new CompletableActorFuture();
+    openLogStream()
+        .onComplete(
+            (log, error) -> {
+              if (error == null) {
+                this.logStream = log;
+                if (!deferredCommitNotifications.isEmpty()) {
+                  deferredCommitNotifications.forEach(Runnable::run);
+                  deferredCommitNotifications.clear();
+                }
+                installDeletionService()
+                    .onComplete(
+                        (deletionService, errorInstall) -> {
+                          if (errorInstall == null) {
+                            installFuture.complete(deletionService);
+                          } else {
+                            installFuture.completeExceptionally(errorInstall);
+                          }
+                        });
+              } else {
+                LOG.error("Failed to install log stream for partition {}", partitionId, error);
+                installFuture.completeExceptionally(error);
+                onFailure();
+              }
+            });
+    return installFuture;
+  }
+
+  private ActorFuture<Void> installDeletionService() {
```

Comment:
It is not only deletion service right? Maybe we split it more or we rename to install state?